server/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -877,11 +877,8 @@ public static boolean isValidActionName(String actionName) {
      */
     @Override
     public void onRequestReceived(long requestId, String action) {
-        try {
-            blockIncomingRequestsLatch.await();
-        } catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-            throw new IllegalStateException("interrupted while waiting for incoming requests block to be removed");
+        if (handleIncomingRequests.get() == false) {
```

Comment:
NIT: maybe just make it a `volatile boolean` since we only use `get` and `set` anyway?